### PR TITLE
feat: support DD_TRACE_ENABLED dynamic configuration

### PR DESCRIPTION
**Pull request by dmehala**

### CODE REVIEW

File: src/datadog/config_manager.h [hunk 1]
```diff
@@ -34,6 +37,8 @@ class ConfigManager {
   // Return the `SpanDefaults` consistent with the most recent configuration.
   std::shared_ptr<const SpanDefaults> get_span_defaults();
 
+  bool get_report_traces();
```
Comment (dmehala) [inline src/datadog/config_manager.h:40]: - [ ] document (and rename?)

**Review by dmehala — COMMENTED** [review]
**TODO:**
- [x] write test
- [ ] ~~remove `null_collector.*`~~ (could be useful)

File: src/datadog/config_manager.h (hunk 1 above)
Comment (dgoffredo) [reply src/datadog/config_manager.h:40]: We could follow the lead of `TracerConfig` and omit the `get_` prefix. It's fine either way.

Yes, please do add a comment here. Maybe:
```
// Return whether spans should be sent to the collector.
```
I don't like the use of the word "should." Something like that, anyway.

File: src/datadog/remote_config.cpp [hunk 2]
```diff
@@ -94,6 +96,15 @@ ConfigUpdate parse_dynamic_config(const nlohmann::json& j) {
     }
   }
 
+  if (auto tracing_enabled_it = j.find("tracing_enabled");
+      tracing_enabled_it != j.cend()) {
+    if (tracing_enabled_it->is_boolean()) {
+      config_update.report_traces = tracing_enabled_it->get<bool>();
+    } else {
+      // TODO: report to telemetry
```
Comment (dgoffredo) [inline src/datadog/remote_config.cpp:104]: coming soon...

**Review by dgoffredo — APPROVED** [review]
:shipit:

File: src/datadog/remote_config.cpp (hunk 2 above)
Comment (dmehala) [reply src/datadog/remote_config.cpp:104]: image

**Pull request merged**